src/EditorFeatures/Core.Wpf/BackgroundWorkIndicator/BackgroundWorkIndicatorScope.cs

```diff
@@ -3,88 +3,49 @@
-
-internal partial class WpfBackgroundWorkIndicatorFactory
+namespace Microsoft.CodeAnalysis.Editor.BackgroundWorkIndicator
 {
-    /// <summary>
-    /// Implementation of an <see cref="IUIThreadOperationScope"/> for the background work indicator. Allows for
-    /// features to create nested work with descriptions/progress that will update the all-up indicator tool-tip
-    /// shown to the user.
-    /// </summary>
-    private sealed class BackgroundWorkIndicatorScope : IUIThreadOperationScope, IProgress<ProgressInfo>
+    internal partial class WpfBackgroundWorkIndicatorFactory
     {
-        private readonly BackgroundWorkIndicatorContext _context;
-
-        // Mutable state of this scope.  Can be mutated by a client, at which point we'll ask our owning context to
-        // update the tooltip accordingly.
-
-        private string _description;
-        private ProgressInfo _progressInfo;
-
-        public IUIThreadOperationContext Context => _context;
-        public IProgress<ProgressInfo> Progress => this;
-
-        public BackgroundWorkIndicatorScope(
-            BackgroundWorkIndicatorContext indicator, string description)
-        {
-            _context = indicator;
-            _description = description;
-        }
-
-        /// <summary>
-        /// Retrieves a threadsafe snapshot of our data for our owning context to use to build the tooltip ui.
-        /// </summary>
-        public (string description, ProgressInfo progressInfo) ReadData_MustBeCalledUnderLock()
-        {
-            Contract.ThrowIfFalse(Monitor.IsEntered(_context.Gate));
-            return (_description, _progressInfo);
-        }
-
-        /// <summary>
-        /// On disposal, just remove ourselves from our parent context.  It will update the UI accordingly.
-        /// </summary>
-        void IDisposable.Dispose()
-            => _context.RemoveScope(this);
-
-        bool IUIThreadOperationScope.AllowCancellation
+        private class BackgroundWorkIndicatorScope : IUIThreadOperationScope
         {
-            get => true;
-            set { }
-        }
+            private readonly BackgroundWorkOperationScope _backgroundScope;
+            private readonly BackgroundWorkIndicatorContext _backgroundWorkIndicatorContext;
 
-        string IUIThreadOperationScope.Description
-        {
-            get
+            public BackgroundWorkIndicatorScope(
+                BackgroundWorkOperationScope backgroundScope,
+                string description,
+                BackgroundWorkIndicatorContext backgroundWorkIndicatorContext)
             {
-                lock (_context.Gate)
-                    return _description;
+                _backgroundScope = backgroundScope;
+                _description = description;
+                _backgroundWorkIndicatorContext = backgroundWorkIndicatorContext;
             }
-            set
+
+            public bool AllowCancellation { get; set; }
```

Comment:
all these props seem odd.  i would expect we would be forwarding to the underlying BackgroundWorkOperationScope  we're wrapping.

Reply:
All part of `IUIThreadOperationScope` :(